jerry-core/jerry.c

```diff
@@ -2087,3 +2087,46 @@ jerry_dispatch_object_free_callback (ecma_external_pointer_t freecb_p, /**< poin
 
   jerry_make_api_available ();
 } /* jerry_dispatch_object_free_callback */
+
+/**
+ * Reserve a slot in the context for a module
+ *
+ * Returns: 0 or positive on success, -1 on failure.
+ */
+
+/* This value is context-independent */
+static int module_index = -1;
```

Comment:
does "context-independent" means all context share the same index? if so, Why?

IMO, the index should be in jerry_context, and each context can have different modules.